CleverTap/Plugins/iOS/CleverTapMessageSender.m

```diff
@@ -0,0 +1,93 @@
+#import "CleverTapMessageSender.h"
+#import "CleverTapUnityCallbackInfo.h"
+#import "CleverTapMessageBuffer.h"
+
+static NSString * kCleverTapGameObjectName = @"IOSCallbackHandler";
+
+@interface CleverTapMessageSender()
+
+@property NSDictionary<CleverTapUnityCallbackInfo *, CleverTapMessageBuffer *> *messageBuffers;
+
+@end
+
+@implementation CleverTapMessageSender
+
++ (instancetype)sharedInstance {
+    static dispatch_once_t once = 0;
+    static id _sharedObject = nil;
+    dispatch_once(&once, ^{
+        _sharedObject = [[self alloc] init];
+    });
+    return _sharedObject;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        self.messageBuffers = [self createBuffers:YES];
+    }
+    return self;
+}
+
+- (NSDictionary<CleverTapUnityCallbackInfo *, CleverTapMessageBuffer *> *)createBuffers:(BOOL)enabled {
+    NSMutableDictionary<CleverTapUnityCallbackInfo *, CleverTapMessageBuffer *> *buffers = [NSMutableDictionary dictionary];
+    NSArray *callbackInfos = [CleverTapUnityCallbackInfo callbackInfos];
+    for (CleverTapUnityCallbackInfo *info in callbackInfos) {
+        if (info.isBufferable) {
+            buffers[info] = [[CleverTapMessageBuffer alloc] initWithEnabled:enabled];
+        }
+    }
+    return [NSDictionary dictionaryWithDictionary:buffers];
+}
+
+- (void)send:(CleverTapUnityCallback)callback withMessage:(NSString *)message {
+    @synchronized (self) {
+        CleverTapUnityCallbackInfo *callbackInfo = [CleverTapUnityCallbackInfo infoForCallback:callback];
+        if (callbackInfo.isBufferable) {
+            CleverTapMessageBuffer *buffer = self.messageBuffers[callbackInfo];
+            if (buffer.isEnabled) {
```

Comment:
Maybe a null check here for buffer, would be more clear than checking isBufferable in the outside if.

Reply:
Updated